bin/tools.py

```diff
@@ -536,7 +543,11 @@ def build_parser():
     runparser.add_argument(
         '--table', action='store_true', help='Print a submissions x testcases table for analysis.'
     )
-    runparser.add_argument('--timeout', type=int, help='Override the default timeout.')
+    runparser.add_argument(
+        '--timeout',
```

Comment:
Hmm; `generate --timeout` and `run --timeout` have the same name, but do different things: the first is a global program timeout, while the second only affects submissions (actually, it probably also affects everything else but that isn't the intention). Maybe we can detangle them. In that case, the `generate --timeout` should probably rename to `generate --global-timeout` or so, but that's quite ugly so not sure about this

Reply:
Yeah, the naming is a bit awkward... we could also rename the `run --timeout` argument to something like `--abort-time`, but that also sounds awkward. I'm also trying to relate this to Kattis' `safety_margin`, but that's a factor, not a hard number (although we could rewrite the running of submissions to use this safety margin instead of a fixed timeout value).

Related to your other comment about rethinking Program timeouts vs. Submission timeouts, we can probably better open a new issue for that for now :stuck_out_tongue: